### PR TITLE
Fix ONNX loading in issues #17516, #17531

### PR DESCRIPTION
**Pull request by sl-sergei**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ x ] I agree to contribute to the project under OpenCV (BSD) License.
- [ x ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ x ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by dkurt** [comment]
Please create a new branch for every pull request:
```
git checkout 3.4
git checkout -b new_branch
```

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 1]
```diff
@@ -902,6 +902,19 @@ void ONNXImporter::populateNet(Net dstNet)
...
+            {
+                Mat inputBuf = getBlob(node_proto, constBlobs, 0);
+
+                LayerParams constParams;
+                constParams.name = node_proto.input(0);
+                constParams.type = "Const";
+                constParams.blobs.push_back(inputBuf);
+
+                opencv_onnx::NodeProto priorsProto;
+                priorsProto.add_output(constParams.name);
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:915]: Why priors?

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 2]
```diff
@@ -902,6 +902,19 @@ void ONNXImporter::populateNet(Net dstNet)
                 Mat bias = getBlob(node_proto, constBlobs, 2);
                 layerParams.blobs.push_back(bias);
             }
+            if (constBlobs.find(node_proto.input(0)) != constBlobs.end())
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:905]: Is that a configuration for GEMM over two constant inputs? Can we use `addConstant`?

File: modules/dnn/src/layers/fully_connected_layer.cpp [hunk 3]
```diff
@@ -116,7 +116,6 @@ class FullyConnectedLayerImpl CV_FINAL : public InnerProductLayer
             CV_CheckEQ(inputs.size(), (size_t)2, "");
             numOutput = inputs[1].back();
             cAxis = inputs[0].size() - 1;
-            CV_CheckEQ(numOutput, inputs[0][cAxis - 1], "");
```
Comment (dkurt) [inline modules/dnn/src/layers/fully_connected_layer.cpp:119]: Please add some test (for reference: https://github.com/opencv/opencv_extra/pull/755 + https://github.com/opencv/opencv/pull/17233)

**Comment by sl-sergei** [comment]
> Please create a new branch for every pull request:
> 
> ```
> git checkout 3.4
> git checkout -b new_branch
> ```

I created a new branch for it in my fork of 3.4.
Should I create a new pull request?

File: modules/dnn/src/onnx/onnx_importer.cpp (hunk 1 above)
Comment (sl-sergei) [reply modules/dnn/src/onnx/onnx_importer.cpp:915]: > Why priors?

That's a typo, sorry, my bad, not sure how did it passed there

File: modules/dnn/src/onnx/onnx_importer.cpp (hunk 2 above)
Comment (sl-sergei) [reply modules/dnn/src/onnx/onnx_importer.cpp:905]: > Is that a configuration for GEMM over two constant inputs? Can we use `addConstant`?

As I see from code in fully_connected_layer.cpp, there are only 2 configurations for GEMM layer, with 1 input and with 2 inputs, so it assumes that there is at least 1 input layer, which is connected in the graph, not constant.

File: modules/dnn/src/layers/fully_connected_layer.cpp (hunk 3 above)
Comment (sl-sergei) [reply modules/dnn/src/layers/fully_connected_layer.cpp:119]: > Please add some test (for reference: [opencv/opencv_extra#755](https://github.com/opencv/opencv_extra/pull/755) + #17233)

Actually, this check makes no sense to me, MatMul operation doesn't require this condition. This check actually does not allow multiplying matrices with dimensions like 12x500 and 500x1000

**Comment by dkurt** [comment]
Yes, please open a new PR

**Pull request closed without merging**